project/api/serializers.py

```diff
@@ -22,25 +25,36 @@ class TerritorySerializer(ModelSerializer):
     """
     Serializes the Territory model as GeoJSON compatible data
     """
+    def to_internal_value(self, data):
+        ret = {}
+
+        # Update ret to include passed in data
+        for field, val in data.items():
+            if field == 'nation':
+                ret['nation'] = Nation.objects.get(pk=val)
+            if field != 'geo' and field != 'nation':
+                ret[field] = val
+
+        # Convert geo field to MultiPolygon if it is a FeatureCollection
+        geojson = loads(data['geo'])
+        if geojson['type'] == 'FeatureCollection':
+            features = geojson['features']
+            features_union = GEOSGeometry(dumps(features[0]['geometry']))
+            features = features[1:]
+
+            for feature in features:
+                if feature['geometry']['type'] == 'Polygon':
+                    features_union = features_union.union(GEOSGeometry(dumps(feature['geometry'])))
```

Comment:
I feel like this operation is very expensive: we're doing n unions!

I'm okay for leaving it like this for now, but it'd be nice to try on some big pieces of data.

Also, this might be relevant: https://stackoverflow.com/questions/15524885/problems-combining-union-multipolygons-in-geodjango

Reply:
Your linked solution would indeed be more efficient if we were working with data that was already in the database. We could add a bunch of temporary Nations, union them, then delete the temporary data, to make use of this approach, but I'm not sure if that would still be faster. Something to test with a large dataset for sure.